### ISSUE TITLE
[BUG] Errors when shooting without a prop protection mod

### ISSUE THREAD

**Issue by shadow7483147** [Stooberton/ACF-3#33]
**Short Description**

Seems like the barrel owner check fails as the CPPIGetOwner method doesn't exist if you don't have a prop protection mod installed. (ie. singleplayer gameplay)

**Additional information (Optional)**

If you spam click enough times in a pod controller, the gun will fire its shot anyway regardless of the check.

**Console Error Logs (Optional)**

```
Wire error (Entity [352][acf_gun] by Player [1][aversion]):
addons/acf-3/lua/entities/acf_gun/init.lua:260: attempt to call method 'CPPIGetOwner' (a nil value)
stack traceback:
	addons/acf-3/lua/entities/acf_gun/init.lua:260: in function 'BarrelCheck'
	addons/acf-3/lua/entities/acf_gun/init.lua:319: in function 'Shoot'
	addons/acf-3/lua/entities/acf_gun/init.lua:287: in function 'AttemptFire'
	addons/acf-3/lua/entities/acf_gun/init.lua:210: in function <addons/acf-3/lua/entities/acf_gun/init.lua:199>
	[C]: in function 'xpcall'
	lua/wire/server/wirelib.lua:58: in function 'TriggerInput'
	lua/wire/server/wirelib.lua:530: in function 'TriggerOutput'
	lua/entities/gmod_wire_pod.lua:294: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function 'Run'
	lua/wire/wireshared.lua:1186: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function <lua/ulib/shared/hook.lua:92>
sv: Wire error (Entity [352][acf_gun] by Player [1][aversion]):
addons/acf-3/lua/entities/acf_gun/init.lua:260: attempt to call method 'CPPIGetOwner' (a nil value)
stack traceback:
	addons/acf-3/lua/entities/acf_gun/init.lua:260: in function 'BarrelCheck'
	addons/acf-3/lua/entities/acf_gun/init.lua:319: in function 'Shoot'
	addons/acf-3/lua/entities/acf_gun/init.lua:287: in function 'AttemptFire'
	addons/acf-3/lua/entities/acf_gun/init.lua:210: in function <addons/acf-3/lua/entities/acf_gun/init.lua:199>
	[C]: in function 'xpcall'
	lua/wire/server/wirelib.lua:58: in function 'TriggerInput'
	lua/wire/server/wirelib.lua:530: in function 'TriggerOutput'
	lua/entities/gmod_wire_pod.lua:294: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function 'Run'
	lua/wire/wireshared.lua:1186: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function <lua/ulib/shared/hook.lua:92>

Wire error (Entity [352][acf_gun] by Player [1][aversion]):
addons/acf-3/lua/entities/acf_gun/init.lua:260: attempt to call method 'CPPIGetOwner' (a nil value)
stack traceback:
	addons/acf-3/lua/entities/acf_gun/init.lua:260: in function 'BarrelCheck'
	addons/acf-3/lua/entities/acf_gun/init.lua:319: in function 'Shoot'
	addons/acf-3/lua/entities/acf_gun/init.lua:287: in function 'AttemptFire'
	addons/acf-3/lua/entities/acf_gun/init.lua:210: in function <addons/acf-3/lua/entities/acf_gun/init.lua:199>
	[C]: in function 'xpcall'
	lua/wire/server/wirelib.lua:58: in function 'TriggerInput'
	lua/wire/server/wirelib.lua:530: in function 'TriggerOutput'
	lua/entities/gmod_wire_pod.lua:294: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function 'Run'
	lua/wire/wireshared.lua:1186: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function <lua/ulib/shared/hook.lua:92>
sv: Wire error (Entity [352][acf_gun] by Player [1][aversion]):
addons/acf-3/lua/entities/acf_gun/init.lua:260: attempt to call method 'CPPIGetOwner' (a nil value)
stack traceback:
	addons/acf-3/lua/entities/acf_gun/init.lua:260: in function 'BarrelCheck'
	addons/acf-3/lua/entities/acf_gun/init.lua:319: in function 'Shoot'
	addons/acf-3/lua/entities/acf_gun/init.lua:287: in function 'AttemptFire'
	addons/acf-3/lua/entities/acf_gun/init.lua:210: in function <addons/acf-3/lua/entities/acf_gun/init.lua:199>
	[C]: in function 'xpcall'
	lua/wire/server/wirelib.lua:58: in function 'TriggerInput'
	lua/wire/server/wirelib.lua:530: in function 'TriggerOutput'
	lua/entities/gmod_wire_pod.lua:294: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function 'Run'
	lua/wire/wireshared.lua:1186: in function 'fn'
	lua/ulib/shared/hook.lua:109: in function <lua/ulib/shared/hook.lua:92>
```

**Comment by Stooberton** [comment]
Install a PP on your client. I recommend nadmod. The dev-branch ReadMe has already been updated to list a PP addon as a dependency. I have no intention of making ACF work without PP, as it will rely more heavily on PP and other systems in the future in order to run better and support more features.

**Comment by TwistedTail** [comment]
While the barrel stuff check depends on CPPI to work, this could be changed to only work only if it exists on the server. Most servers using ACF will be using a prop protection addon, so this issue will appear either on SP/P2P games where you will most likely not need for prop protection. After all, servers without PP will have greater issues than people barrel stuffing.

**Comment by TwistedTail** [comment]
Fixed on 84a37ed.
